src/Net/Http.fs

```diff
@@ -303,34 +314,654 @@ module HttpContentTypes =
+            (".wmp", "video/x-ms-wmp")
+            (".wmv", "video/x-ms-wmv")
+            (".wmx", "video/x-ms-wmx")
+            (".wmz", "application/x-ms-wmz")
+            (".woff", "application/font-woff")
+            (".wpl", "application/vnd.ms-wpl")
+            (".wps", "application/vnd.ms-works")
+            (".wri", "application/x-mswrite")
+            (".wrl", "x-world/x-vrml")
+            (".wrz", "x-world/x-vrml")
+            (".wsc", "text/scriptlet")
+            (".wsdl", "text/xml")
+            (".wvx", "video/x-ms-wvx")
+            (".x", "application/directx")
+            (".xaf", "x-world/x-vrml")
+            (".xaml", "application/xaml+xml")
+            (".xap", "application/x-silverlight-app")
+            (".xbap", "application/x-ms-xbap")
+            (".xbm", "image/x-xbitmap")
+            (".xdr", "text/plain")
+            (".xht", "application/xhtml+xml")
+            (".xhtml", "application/xhtml+xml")
+            (".xla", "application/vnd.ms-excel")
+            (".xlam", "application/vnd.ms-excel.addin.macroEnabled.12")
+            (".xlc", "application/vnd.ms-excel")
+            (".xld", "application/vnd.ms-excel")
+            (".xlk", "application/vnd.ms-excel")
+            (".xll", "application/vnd.ms-excel")
+            (".xlm", "application/vnd.ms-excel")
+            (".xls", "application/vnd.ms-excel")
+            (".xlsb", "application/vnd.ms-excel.sheet.binary.macroEnabled.12")
+            (".xlsm", "application/vnd.ms-excel.sheet.macroEnabled.12")
+            (".xlsx", "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet")
+            (".xlt", "application/vnd.ms-excel")
+            (".xltm", "application/vnd.ms-excel.template.macroEnabled.12")
+            (".xltx", "application/vnd.openxmlformats-officedocument.spreadsheetml.template")
+            (".xlw", "application/vnd.ms-excel")
+            (".xml", "text/xml")
+            (".xmta", "application/xml")
+            (".xof", "x-world/x-vrml")
+            (".XOML", "text/plain")
+            (".xpm", "image/x-xpixmap")
+            (".xps", "application/vnd.ms-xpsdocument")
+            (".xrm-ms", "text/xml")
+            (".xsc", "application/xml")
+            (".xsd", "text/xml")
+            (".xsf", "text/xml")
+            (".xsl", "text/xml")
+            (".xslt", "text/xml")
+            (".xsn", "application/octet-stream")
+            (".xss", "application/xml")
+            (".xspf", "application/xspf+xml")
+            (".xtp", "application/octet-stream")
+            (".xwd", "image/x-xwindowdump")
+            (".z", "application/x-compress")
+            (".zip", "application/zip") |]
+    let private map =
+        let d = Dictionary<string,string>(StringComparer.OrdinalIgnoreCase)
+        for (k,v) in pairs do
+            d.Add(k,v)
+        d
+    let tryFind ext =
+        match map.TryGetValue(ext) with
+        | true, mime -> Some mime
+        | _ -> None
+
```

Comment:
Suggestion (not mandatory) - replace this part with the following

      let private map = Map.ofArray pairs
      let tryFind ext = let tryFind (ext: string) = map.TryFind(ext.ToLowerInvariant())

and then you can also remove the import of System.Collections.Generic